parallaxinc/solo
# Refactor custom code block

Reposted because the original PR had too many conflicts.

Addresses #128

 - Removes original field_code custom field type
 - Adds improved field_ace based on new core custom fields APIs and examples shown at Blockly Dev conference
 - Refactors custom code block to eliminate hidden fields (see #113)

To Test:
 - create a custom code block and expand it to edit it.
 - populate all fields
 - minimize the block
 - use the code view to ensure that text added to fields appears correctly in the code
 - Save the project with the block expanded, reopen and verify
 - Save the project with the block minimized, reopen and verify
 - Add inputs
 - add blocks to those inputs
 - ensure that when adding/changing the number of inputs, blocks stay attached appropriately
 - Ensure attached block values render correctly in the c code (use the '@' replacers)
 - Open older files with custom code blocks to ensure they render correctly
